src/attr/_make.py

```diff
@@ -597,6 +599,48 @@ def _transform_attrs(
     return _Attributes((AttrsClass(attrs), base_attrs, base_attr_map))
 
 
+def _make_cached_property_getattr(
+    cached_properties,
+    cls,
+):
+    lines = [
+        # Wrapped to get `__class__` into closure cell for super()
+        # (It will be replaced with the newly constructed class after construction).
+        "def wrapper(_cls, cached_properties, _cached_setattr_get):",
+        "    __class__ = _cls",
+        "    def __getattr__(self, item):",
+        "         func = cached_properties.get(item)",
+        "         if func is not None:",
+        "              result = func(self)",
+        "              _setter = _cached_setattr_get(self)",
+        "              _setter(item, result)",
+        "              return result",
+        "         if '__attrs_original_getattr__' in vars(__class__):",
+        "              return __class__.__attrs_original_getattr__(self, item)",
```

Comment:
Do you have to do this check here? Can this be checked at class construction time and passed as a parameter to `_make_cached_property_getattr`? Might make the `__class__` global unnecessary too.

Reply:
I've moved the check out to construction, and passed in the `original_getattr` to avoid the class lookup.

`__class__` is still needed in the closure for the `super()` call to work.

I've removed  `__attrs_original_getattr__` on the class, as a separate MR, because it needed some other changes  for it goes through the `__class__` closure reference updating, which has it's own trade-off.